src/hardware/vga_other.cpp

```diff
@@ -139,21 +148,21 @@ static Bitu read_crtc_data_other(Bitu /*port*/,Bitu /*iolen*/) {
 	case 0x0B:	/* Cursor End Register */
 		return vga.other.cursor_end;
 	case 0x0C:	/* Start Address High Register */
-		return (Bit8u)(vga.config.display_start >> 8);
+		return static_cast<uint8_t>(vga.config.display_start >> 8);
```

Comment:
I don't think these `static_casts` actually improve anything (because function returns uint8_t already)… Maybe they serve as documentation, that's all…

Reply:
It's unfortunately more `Bitu` baggage.  These member variables are a whopping 64-bit.. but I believe should be limited to 16-bit.

``` c++
/* Video drawing */
        Bitu display_start;
        Bitu real_start;
        bool retrace;                                   /* A retrace is active */
        Bitu scan_len;
        Bitu cursor_start;

```

So the downshift by 8 still results in a 56-bit value, so Clang flags it (on my side.. where I have more warnings turned on):

![2021-07-08_19-01](https://user-images.githubusercontent.com/1557255/125012340-e032d300-e01e-11eb-98ef-d86b1dee9c65.png)

I'd like to make these types correct, size-wise, but makes sense to do that in a VGA-register sizing PR.

Reply:
Ah, compiler complains about implicit conversion here - I agree, it makes sense.